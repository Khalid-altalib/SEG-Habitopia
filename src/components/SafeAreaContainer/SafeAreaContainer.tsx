// React Native
import { StyleSheet, View } from "react-native";

// React Native Safe Area Context
import {
  SafeAreaProvider,
  useSafeAreaInsets,
} from "react-native-safe-area-context";

export type SafeAreaContainerProps = {
  /** The children to show within the container. */
  children: any;
};

/**
 * A container that ensures that all of its children are within the "safe area" of the screen
 * and not obscured by issues such as the iPhone notch.
 *
 * @param props - The children components to be displayed within this safe area container.
 * @returns - The container with its children.
 */
const SafeAreaContainer = (props: SafeAreaContainerProps) => {
  const insets = useSafeAreaInsets();

  const styles = StyleSheet.create({
    safeAreaPadding: {
      paddingTop: insets.top,
      paddingBottom: insets.bottom,
      paddingLeft: insets.left,
      paddingRight: insets.right,
    },
  });

  return (
    <SafeAreaProvider>
      <View style={styles.safeAreaPadding}>{props.children}</View>
    </SafeAreaProvider>
  );
};

export default SafeAreaContainer;
