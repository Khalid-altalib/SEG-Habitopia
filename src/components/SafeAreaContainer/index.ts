import SafeAreaContainer from "./SafeAreaContainer";

export default SafeAreaContainer;
