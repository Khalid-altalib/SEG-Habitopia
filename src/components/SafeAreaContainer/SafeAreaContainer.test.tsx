// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// React Native Safe Area Context
import { SafeAreaProvider } from "react-native-safe-area-context";

// Habitopia
import SafeAreaContainer from "@components/SafeAreaContainer";

it("Renders correctly.", () => {
  render(
    <SafeAreaProvider>
      <SafeAreaContainer>This is some example content.</SafeAreaContainer>
    </SafeAreaProvider>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
