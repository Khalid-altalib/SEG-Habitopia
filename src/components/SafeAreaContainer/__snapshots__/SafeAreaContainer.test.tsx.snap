// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
